### PR TITLE
[Feature Proposal] Adding SpriteSheet class

### PR DESCRIPTION
**Pull request by erickzanardo**
This PR is a Feature Proposal to add a SpriteSheet class.

This class would serve as a helper for the already existent Sprite class. This just an initial proposal, we can improve this adding more methods.

If accepted, I can add the missing documentation and example.

### CODE REVIEW

File: lib/spritesheet.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      final spriteList = spriteRow.sublist(from, to);
+
+      return Animation.spriteList(
+        spriteList,
+        stepTime: stepTime,
+        loop: loop,
+      );
+    }
+
+    return null;
```
Comment (luanpotter) [inline lib/spritesheet.dart]: better throw an error, don't you think?

**Comment by erickzanardo** [comment]
@luanpotter I have updated this PR, can you take another look please?

File: lib/spritesheet.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    final Sprite s = _sprites[row][column];
+
+    assert(s != null, 'No sprite found for row $row and column $column');
+
+    return s;
+  }
+
+  /// Creates an animation from this SpriteSheet
+  ///
+  /// An [from]  and a [to]  parameter can be specified to create an animation from a subset of the columns on the row
```
Comment (luanpotter) [inline lib/spritesheet.dart]: double spaces after <[from]>

File: lib/spritesheet.dart [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import 'sprite.dart';
+import 'animation.dart';
+
```
Comment (luanpotter) [inline lib/spritesheet.dart]: add a dart doc for the class :D

File: doc/images.md [hunk 4]
```diff
@@ -196,3 +196,31 @@ class MyGame extends Game {
...
+
+Now you can use the animation directly or use it in an animation component.
+
+You can also get a single frame of the sprite sheet using the `getSprite` method:
+
+```dart
+spritesheet.getSprite(0, 0) // row, column;
+```
+
+You can see a full example of the SpriteSheet class [here](doc/examples/spritesheet)
```
Comment (luanpotter) [inline doc/images.md]: period**.**

File: lib/spritesheet.dart [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    assert(s != null, 'No sprite found for row $row and column $column');
+
+    return s;
+  }
+
+  /// Creates an animation from this SpriteSheet
+  ///
+  /// An [from]  and a [to]  parameter can be specified to create an animation from a subset of the columns on the row
+  Animation createAnimation(int row,
+      {double stepTime, loop = true, int from = 0, int to}) {
```
Comment (luanpotter) [inline lib/spritesheet.dart]: add type to all parameters (loop)

**Comment by erickzanardo** [comment]
@luanpotter all done!

**Pull request merged**